h2o-core/src/main/java/water/fvec/Vec.java

```diff
@@ -394,7 +394,7 @@ public static Vec makeCon(double x, long len, int log_rows_per_chunk, boolean re
 
   public static Vec makeCon(double x, long len, int log_rows_per_chunk, boolean redistribute, byte type) {
     int chunks0 = (int)Math.max(1,len>>log_rows_per_chunk); // redistribute = false
-    int chunks1 = (int)Math.min( 4 * H2O.NUMCPUS * H2O.CLOUD.size(), len); // redistribute = true
+    int chunks1 = (int)Math.min( 4 * H2O.ARGS.nthreads * H2O.CLOUD.size(), len); // redistribute = true
```

Comment:
I am not sure if user should have a way to indirectly influence number of vectors in a chunk by settings number of threads available. We surely use it in SQLManager, where number of connections in the pool is derived from the number of threads available, but using this in a Vec is a **big change** and may affect performance greatly, doesn't it ?'
If this is something to change, I'd do it myself in a separate PR with a proper issue and some performance tests before merging it.

Reply:
I understand the concern, but the number of chunks is deeply connected with parallelism when manipulating the vector, right? and parallelism is associated with nthreads, not its default NUMCPUS, so why was the non-configurable NUMCPUS chosen here in the first place.